build/lpc55pins/src/lib.rs

```diff
@@ -0,0 +1,156 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::Result;
+use proc_macro2::TokenStream;
+use quote::{format_ident, ToTokens, TokenStreamExt};
+use serde::Deserialize;
+use std::io::Write;
+
+#[derive(Clone, Debug, Deserialize)]
+struct Pin {
+    port: usize,
+    pin: usize,
+}
+
+impl Pin {
+    fn get_port_pin(&self) -> (usize, usize) {
+        if self.pin > 32 {
```

Comment:
This could be a one-liner: `assert!(self.pin <= 32, "Invalid pin {}", self.pin)` (same for the port check below)

I'm skeptical that `32` is a valid pin number, so maybe it should be
```
assert!(self.pin < 32, "Invalid pin {}", self.pin)
```

Reply:
asserts are usually expensive because of debug formatting so I tend to shy away from using them. You're correct about the pin number though.